IpmiFeaturePkg/Samples/IpmiStatusReporter/IpmiStatusReporter.c

```diff
@@ -0,0 +1,62 @@
+/** @file
+  Sample common code for sending EfiReportStatus to IPMI SEL.
+
+  Copyright (c) Microsoft Corporation
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <IpmiStatusReporter.h>
+
+/**
+  Sample implementation for logging a Status Code event to the System Event Log (SEL).
+
+  @param[in]  CodeType   The status code type.
+  @param[in]  Value      The status code value.
+  @param[in]  Instance   The status code instance number.
+  @param[in]  CallerId   Pointer to a GUID that identifies the caller of this
+                         function. This is an optional parameter that may be NULL.
+  @param[in]  Data       Pointer to the extended data buffer. This is an optional
+                         parameter that may be NULL.
+
+  @retval     EFI Status Code.
+**/
+EFI_STATUS
+EFIAPI
+IpmiReportStatusCode (
+  IN EFI_STATUS_CODE_TYPE   CodeType,
+  IN EFI_STATUS_CODE_VALUE  Value,
+  IN UINT32                 Instance,
```

Comment:
Instance isn't used, but would presumably be used to say this status code was hit by instance # X? Perhaps adding a comment about how this parameter might be used or dropping the param would make sense.

Reply:
Added more to the comment, frankly even the PI spec is somewhat vague here to its probably very status code specific.